### PR TITLE
Feat/get nearby drivers api

### PR DESCRIPTION
**Pull request by nikhilleo10**
### Ticket Link
Get nearby drivers API
---

### Related Links

---

### Description
- Driver DAOs 
- Driver Routes
- Driver dao tests
- Driver route tests
---

### Steps to Reproduce / Test

---

---

### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's

---

### CODE REVIEW

File: lib/daos/driverDao.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { models } from '@models';
+import { badImplementation } from '@utils/responseInterceptors';
+import sequelize from 'sequelize';
+
+export const findOneUser = async () => models.users.findOne();
```
Comment (aseer-ws) [inline lib/daos/driverDao.js:5]: Can't we remove this?

File: lib/daos/tests/driverDao.test.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import { resetAndMockDB } from '@utils/testUtils';
+import { sample } from 'lodash';
+import sequelize from 'sequelize';
+
+describe('Driver daos', () => {
```
Comment (aseer-ws) [inline lib/daos/tests/driverDao.test.js]: Lets keep the name `driverDao`

File: lib/daos/tests/driverDao.test.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  describe('Find nearby drivers', () => {
```
Comment (aseer-ws) [inline lib/daos/tests/driverDao.test.js]: Let keep the name of function `getNearbyDrivers`

File: lib/models/index.js [hunk 4]
```diff
@@ -1,33 +1,37 @@
...
 export const models = {
-  oauthAccessTokens: oauthAccessTokens(sequelize, Sequelize.DataTypes),
-  oauthClientResources: oauthClientResources(sequelize, Sequelize.DataTypes),
-  oauthClientScopes: oauthClientScopes(sequelize, Sequelize.DataTypes),
-  oauthClients: oauthClients(sequelize, Sequelize.DataTypes),
-  users: users(sequelize, Sequelize.DataTypes),
+  users: user(sequelize, Sequelize.DataTypes),
+  customer: customer(sequelize, Sequelize.DataTypes),
+  driver: driver(sequelize, Sequelize.DataTypes),
+  vehicle: vehicle(sequelize, Sequelize.DataTypes),
```
Comment (aseer-ws) [inline lib/models/index.js]: can we keep model names as plural

File: lib/routes/drivers/tests/routes.test.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
+import { resetAndMockDB } from '@utils/testUtils';
+import { mockData } from '@utils/mockData';
+import _ from 'lodash';
+
+const { MOCK_VEHICLE: vehicle } = mockData;
+
+function getUrl(location, page, typeOfVehicle, limit) {
+  return `/drivers/nearby?location=${location}&page=${page}&typeOfVehicle=${typeOfVehicle}&limit=${limit}`
```
Comment (aseer-ws) [inline lib/routes/drivers/tests/routes.test.js]: Let's use `qs` library for this.

File: lib/routes/drivers/tests/routes.test.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+const limit = 10;
+const typeOfVehicle = _.sample(['AUTO','BIKE','CAR']);
+const location = '26.1564,75.6813';
+const page = 1;
+
+describe('/driver route tests ', () => {
```
Comment (aseer-ws) [inline lib/routes/drivers/tests/routes.test.js]:
```suggestion
describe('/drivers/nearby route tests ', () => {
```

File: utils/validationUtils.js [hunk 7]
```diff
@@ -58,3 +58,14 @@ export const oneOfAllowedScopes = Joi.string()
...
+
+export const locationSchema = JoiBase.string().custom((val, helper) => {
+  const coordinates = val.split(',');
+  if((coordinates[1] >= -90 && coordinates[1] <= 90) && (coordinates[0] >= -180 && coordinates[0] <= 180)) {
+    return val
+  }
+  return helper.message('Invalid Coordinates');
+}).error(new Error('Invalid Coordinates.'))
+
+export const typeOfVehicleSchema = Joi.string().valid(...['AUTO','BIKE','CAR']).required().error(new Error('Invalid type of vehicle.'))
```
Comment (aseer-ws) [inline utils/validationUtils.js]: Wherever we using the vehicle type `['AUTO', 'BIKE', 'CAR']`, lets bring it from a constant

**Review by aseer-ws — CHANGES_REQUESTED** [review]

File: lib/daos/driverDao.js (hunk 1 above)
Comment (nikhilleo10) [reply lib/daos/driverDao.js:5]: If I remove this. We will have to remove it from many other places so I didn't removed this.

**Review by aseer-ws — APPROVED** [review]

**Pull request merged**